File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/Orders/Events/OrderPlaced.cs
using System;
using ENode.Eventing;

namespace Registration.Orders
{
    public class OrderPlaced : DomainEvent<Guid>
    {
        public Guid ConferenceId { get; private set; }
        public OrderTotal OrderTotal { get; private set; }
        public DateTime ReservationExpirationDate { get; private set; }
        public string AccessCode { get; private set; }

        public OrderPlaced() { }
        public OrderPlaced(Guid conferenceId, OrderTotal orderTotal, DateTime reservationExpirationDate, string accessCode)
        {
            ConferenceId = conferenceId;
            OrderTotal = orderTotal;
            ReservationExpirationDate = reservationExpirationDate;
            AccessCode = accessCode;
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/orders/events/OrderPlaced.java
package com.microsoft.conference.registration.domain.orders.events;

import com.microsoft.conference.registration.domain.orders.models.OrderTotal;
import org.enodeframework.eventing.DomainEvent;

import java.util.Date;

public class OrderPlaced extends DomainEvent<String> {
    public String conferenceId;
    public OrderTotal orderTotal;
    public Date reservationExpirationDate;
    public String accessCode;

    public OrderPlaced() {
    }

    public OrderPlaced(String conferenceId, OrderTotal orderTotal, Date reservationExpirationDate, String accessCode) {
        this.conferenceId = conferenceId;
        this.orderTotal = orderTotal;
        this.reservationExpirationDate = reservationExpirationDate;
        this.accessCode = accessCode;
    }
}
